### PR TITLE
[stdlib] convert `withUnsafeMutablePointer()` to typed throws

### PR DESCRIPTION
**Pull request by glessard**
Part of the ongoing conversion of standard library rethrows functions, after [SE-0413](https://github.com/apple/swift-evolution/blob/main/proposals/0413-typed-throws.md).

This converts the top-level `withUnsafeMutablePointer(to:_:)` function.
Its corresponding no-stack-protection equivalent is omitted here, due to a compiler issue (rdar://121289127).

### CODE REVIEW

**Comment by glessard** [comment]
@swift-ci please test

**Comment by glessard** [comment]
@swift-ci please test

**Review by Azoy — APPROVED** [review]
LGTM

**Comment by glessard** [comment]
@swift-ci please test

**Comment by glessard** [comment]
The previous failures should be fixed by https://github.com/apple/swift/pull/72152

File: test/api-digester/stability-stdlib-abi-without-asserts.test [hunk 1]
```diff
@@ -105,6 +105,9 @@ Constructor Result.init(__abi_catching:) is a new API without @available attribu
 Constructor Result.init(catching:) has been removed
 Func Result.get() has been renamed to Func __abi_get()
 Func Result.get() has mangled name changing from 'Swift.Result.get() throws -> A' to 'Swift.Result.__abi_get() throws -> A'
+Func withUnsafeMutablePointer(to:_:) has been renamed to Func __abi_se0413_withUnsafeMutablePointer(to:_:)
+Func withUnsafeMutablePointer(to:_:) has mangled name changing from 'Swift.withUnsafeMutablePointer<A, B>(to: A, _: (Swift.UnsafeMutablePointer<A>) throws -> B) throws -> B' to 'Swift.__abi_se0413_withUnsafeMutablePointer<A, B>(to: A, _: (Swift.UnsafeMutablePointer<A>) throws -> B) throws -> B'
```
Comment (glessard) [inline test/api-digester/stability-stdlib-abi-without-asserts.test:109]:
```suggestion
Func withUnsafeMutablePointer(to:_:) has mangled name changing from 'Swift.withUnsafeMutablePointer<A, B>(to: inout A, _: (Swift.UnsafeMutablePointer<A>) throws -> B) throws -> B' to 'Swift.__abi_se0413_withUnsafeMutablePointer<A, B>(to: inout A, _: (Swift.UnsafeMutablePointer<A>) throws -> B) throws -> B'
```

**Comment by glessard** [comment]
@swift-ci please test

**Pull request merged**